src/VecSim/algorithms/svs/svs_tiered.h

```diff
@@ -0,0 +1,726 @@
+            return this->frontendIndex->isLabelExists(label);
+        }();
+
+        if (label_exists) {
+            std::scoped_lock lock(this->flatIndexGuard, this->mainIndexGuard, this->journal_mutex);
+            if (this->frontendIndex->isLabelExists(label)) {
+                ret = this->frontendIndex->deleteVector(label);
+                assert(ret == 1 && "unexpected deleteVector result");
+                journal.emplace_back(label, false);
+            }
+            ret += svs_index->deleteVectors(&label, 1);
+            assert(ret < 2 && "deleteVector: vector duplication in both indices");
+        } else {
+            std::scoped_lock lock(this->mainIndexGuard);
+            ret += svs_index->deleteVectors(&label, 1);
+        }
+        return ret;
+    }
+
+    size_t indexSize() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexSize() + this->backendIndex->indexSize();
+    }
+
+    size_t indexLabelCount() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexLabelCount() + this->backendIndex->indexLabelCount();
+    }
+    size_t indexCapacity() const override {
+        std::shared_lock<std::shared_mutex> flat_lock(this->flatIndexGuard);
+        std::shared_lock<std::shared_mutex> main_lock(this->mainIndexGuard);
+        return this->frontendIndex->indexCapacity() + this->backendIndex->indexCapacity();
+    }
+
+    double getDistanceFrom_Unsafe(labelType label, const void *blob) const override {
+        double flat_dist = std::numeric_limits<double>::quiet_NaN();
+        {
+            std::shared_lock<decltype(this->flatIndexGuard)> lock(this->flatIndexGuard);
+            flat_dist = this->frontendIndex->getDistanceFrom_Unsafe(label, blob);
+        }
+        if (!std::isnan(flat_dist)) {
+            return flat_dist;
+        } else {
+            std::shared_lock<decltype(this->mainIndexGuard)> lock(this->mainIndexGuard);
+            return this->backendIndex->getDistanceFrom_Unsafe(label, blob);
+        }
+    }
+
+    VecSimIndexDebugInfo debugInfo() const override {
+        auto info = Base::debugInfo();
+        return info;
+    }
+
+    VecSimIndexBasicInfo basicInfo() const override {
+        VecSimIndexBasicInfo info = this->backendIndex->getBasicInfo();
+        info.blockSize = info.blockSize;
+        info.isTiered = true;
+        info.algo = VecSimAlgo_SVS;
+        return info;
+    }
+
+    VecSimDebugInfoIterator *debugInfoIterator() const override {
+        //  Get the base tiered fields.
+        auto *infoIterator = Base::debugInfoIterator();
+        return infoIterator;
+    }
+
+    VecSimBatchIterator *newBatchIterator(const void *queryBlob,
```

Comment:
We have recently made changes to how we manage the query allocation for batch iterator.
No need to copy it here, and TieredSVS_BatchIterator, there is no need to pass it the VecSimBatchIterator base class (it is not used)
see https://github.com/RedisAI/VectorSimilarity/pull/665/files#diff-6d14fc9981d2d787876160c65e148983494241bc7e98c1d7280e7019eeab7510

( i have already done the related fix in svs.h in the same pr :):
https://github.com/RedisAI/VectorSimilarity/pull/665/files#diff-1627c632037ef1f31507a00ce498450cc230ba0215fc486541246285ba3e9729
)

Reply:
Done.